### ISSUE TITLE
dylib can't be found for debug on m1 osx

### ISSUE THREAD

**Issue by Hyiker** [xmake-io/xmake#2345]
### Xmake Version

2.6.5

### Operating System Version and Architecture

macOS Monterey 12.3.1, m1 pro

### Describe Bug

I was trying to debug my project with some third party package, but llvm ends up telling me it can't find dylibs at the very beginning:

```bash
xmake r -d
(lldb) target create ".../build/macosx/arm64/debug/TrivialRenderer"
Current executable set to '.../build/macosx/arm64/debug/TrivialRenderer' (arm64).
(lldb) r
Process 1368 launched: '.../build/macosx/arm64/debug/TrivialRenderer' (arm64)
dyld[1368]: Library not loaded: @rpath/libembree3.3.dylib
  Referenced from: .../build/macosx/arm64/debug/TrivialRenderer
  Reason: tried: '/usr/local/lib/libembree3.3.dylib' (no such file), '/usr/lib/libembree3.3.dylib' (no such file)
```

However, I can successfully run the target with `xmake r TrivialRenderer`, and I can debug the project without the error on my Windows 11 x86_64 platform.

Additionally, this problem only occurs when I add packages with dylibs, nothing goes wrong if  I'm debugging a one without them.

### Expected Behavior

The llvm debugger should run successfully.

### Project Configuration

```lua
add_rules("mode.debug", "mode.release")


package("pbrt-parser")
    add_deps("cmake")
    set_sourcedir(path.join(os.scriptdir(), "ext/pbrt-parser"))
    on_install(function (package)
        local configs = {}
        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
        import("package.tools.cmake").install(package, configs)
    end)
package_end()

add_requires("embree 3.13.3", "glog 0.5.0", "tinyexr 1.0.1", "pbrt-parser")


target("TrivialRenderer")
    set_kind("binary")
    add_files("src/**.cpp", {includedirs="src/"})
    set_languages("c17", "cxx20")
    add_packages("embree", "glog", "tinyexr", "pbrt-parser")
    add_defines("PRECISION_32", "_CRT_SECURE_NO_WARNINGS")
```

**Comment by waruqi** [comment]
run `otool -L targetfile` and check rpath

**Comment by Hyiker** [comment]
> run `otool -L targetfile` and check rpath

Thanks, I've solved this problem by running `install_name_tool -add_rpath` to add rpath for the binary executable. However, I wonder if xmake can do this automatically during compiling when I got remote packages installed(like tbb). Remote packages may installed in a path like `~/.xmake/packages/t/tbb/2021.5.0/02686119803f4d2fad211638d224dbec/lib` with hashes, can I get it with a built-in variable for `add_rpathdirs `? Or, I just have to maually(or with a script) run `install_name_tool`  every time I recompile the project?
